### PR TITLE
use boost:::threads for png creation

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- `CMakeLists.txt`: add `thread` to boost requirements (#1251)
- add method `join()` to
  - `PngCreator`
  - `LiveViewClient`
- `PngCreator.hpp`: use `boos::threads` for asynchronous png creation

- [x] Tests with the current dev
- [x] rebase against #1251

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
@ax3l  Something is wrong with your boost version on the compile node. We can check this next week.

**Comment by psychocoderHPC** [comment]
Might be that this is a nvcc bug: see https://svn.boost.org/trac/boost/ticket/10625
I will check this next week.

**Comment by ax3l** [comment]
we can try if boost 1.56.0+ avoids the problem and upgrade if necessary. By that I could also remove the other old [nvcc/boost 1.55.0 work around](https://github.com/ComputationalRadiationPhysics/picongpu/blob/4d05eea911111fb4d0a7e7ef7cd394feccbd939d/src/picongpu/CMakeLists.txt#L247-L255)

**Comment by psychocoderHPC** [comment]
I rebased this plugin and compile all on our cluster hypnos without any problems.
I used cuda 6.0 and boost/1.55.0 on hypnos. I can't see where the problem on the compile suite is coming from.

**Comment by ax3l** [comment]
what version of gcc did you use? suite uses 4.6.4

**Comment by ax3l** [comment]
we can try gcc 4.6.4 + boost 1.52/56 tomorrow, having `thread` enabled during compile this time

**Comment by psychocoderHPC** [comment]
I checked my settuing on hypnos, I used gcc 4.6.2.
Currently Loaded Modulefiles:
  1) mpfr/3.0.0                    8) boost/1.56.0 and 1.55.0
  2) mpc/0.8.2                     9) hdf5-parallel/1.8.14
  3) gmp/4.3.1                    10) libsplash/1.2.4
  4) gcc/4.6.2                    11) numactl/2.0.7
  5) cmake/3.0.1                  12) libmxml/2.8
  6) cuda/7.0                     13) adios/1.9.0
  7) openmpi/1.8.4.kepler.cuda70  14) valgrind/3.8.1

**Comment by ax3l** [comment]
re-try with boost 1.56.0 instead of 1.55.0

**Comment by psychocoderHPC** [comment]
Ohh it runs, should I change the minimum boost requirement

**Comment by ax3l** [comment]
yes, pls increase (CMakeLists.txt, INSTALL.md)

**Comment by ax3l** [comment]
will do in a follow-up

**Comment by psychocoderHPC** [comment]
ready for review

**Comment by erikzenker** [comment]
This header is crying: "Please write some documentation for me!"

File: src/picongpu/include/plugins/output/images/LiveViewClient.hpp [hunk 1]
```diff
@@ -76,7 +85,7 @@ namespace picongpu
     inline void LiveViewClient::operator() < DataBox<PitchedBox<float3_X, DIM2 > > >(
                                                                                    const DataBox<PitchedBox<float3_X, DIM2 > > data,
```
Comment (erikzenker) [inline src/picongpu/include/plugins/output/images/LiveViewClient.hpp]: Is there a coding guideline for function parameter indentation and alignment ?

Comment (psychocoderHPC) [inline src/picongpu/include/plugins/output/images/LiveViewClient.hpp]: No currently not :-(

Comment (ax3l) [inline src/picongpu/include/plugins/output/images/LiveViewClient.hpp]: "not as ugly as this one"

Comment (ax3l) [inline src/picongpu/include/plugins/output/images/LiveViewClient.hpp]: Reference: [wiki page](https://github.com/ComputationalRadiationPhysics/picongpu/wiki/Coding-Guide-Lines)

Comment (psychocoderHPC) [inline src/picongpu/include/plugins/output/images/LiveViewClient.hpp]: The current coding style not define if we indent only or align function parameters!

But I will change this line.

**Comment by ax3l** [comment]
@erikzenker pls review

**Comment by erikzenker** [comment]
Review finished, well some documentation would help me to understand what this code is actually doing. For me it looks like it does what it is indented for.

**Comment by erikzenker** [comment]
Ready to merge, objections ?

**Comment by ax3l** [comment]
great for me, but I did not take the time to review the implementation in detail.

**Comment by psychocoderHPC** [comment]
push the :radio_button:

**Comment by erikzenker** [comment]
Good work :hand: --> :radio_button:

**Pull request merged**